### PR TITLE
supporting creditcard account_type

### PR DESCRIPTION
**Pull request by hhemanth**
@ouranos 
Added logic to identify creditcard account and suitably populate acct_type field.

### CODE REVIEW

File: lib/ofx/parser/ofx220.rb [hunk 1]
```diff
@@ -16,9 +16,14 @@ def build_account_info(node)
           description:       node.search('desc').inner_text.upcase,
           bank_id:           nested_account_info.search('bankid').inner_text,
           id:                nested_account_info.search('acctid').inner_text,
-          type:              ACCOUNT_TYPES[node.search('accttype').inner_text.to_s.upcase],
+          type:              fetch_account_type(node),
         })
       end
+
+      def fetch_account_type(node)
```
Comment (ouranos) [inline lib/ofx/parser/ofx220.rb:23]: As mentioned on #1 I believe this is redundant.

File: lib/ofx/parser/ofx102.rb [hunk 2]
```diff
@@ -61,14 +62,27 @@ def build_account(node)
...
           :currency          => node.search("stmtrs > curdef, ccstmtrs > curdef").inner_text
         })
       end
 
+      def fetch_account_type(node)
+        acct_type = ACCOUNT_TYPES[node.search("bankacctfrom > accttype").inner_text.to_s.upcase]
+        acct_type ||= node.search('ccacctinfo').any? ? :creditcard : nil
+      end
+
+      def build_status(node)
```
Comment (ouranos) [inline lib/ofx/parser/ofx102.rb]: This shouldn't be there

**Review by ouranos — COMMENTED** [review]
This is also missing the extd name that you pushed to #1?

File: lib/ofx/parser/ofx220.rb (hunk 1 above)
Comment (hhemanth) [reply lib/ofx/parser/ofx220.rb:23]: Its needed as the implementation is slightly different

**Comment by hhemanth** [comment]
@ouranos 
I have addressed the comments and I have added specs

**Comment by hhemanth** [comment]
This branch pulled from feature/account_discovery which is way behind and is breaking all functionality and specs in marlin_banking_adapter. But the branch from staging (feature/MARLIN-438-supporting-new-acct-types) is working fine.

**Review by ouranos — APPROVED** [review]

File: lib/ofx/parser/ofx102.rb [hunk 3]
```diff
@@ -61,14 +62,19 @@ def build_account(node)
...
-          :type              => ACCOUNT_TYPES[node.search("bankacctfrom > accttype").inner_text.to_s.upcase],
+          :type              => fetch_account_type(node),
           :transactions      => build_transactions(node),
           :balance           => build_balance(node),
           :available_balance => build_available_balance(node),
           :currency          => node.search("stmtrs > curdef, ccstmtrs > curdef").inner_text
         })
       end
 
+      def fetch_account_type(node)
```
Comment (iseessel) [inline lib/ofx/parser/ofx102.rb:73]: Can you explain this to me? Might be useful to have comment.

File: spec/ofx/ofx102_spec.rb [hunk 4]
```diff
@@ -42,5 +42,40 @@
...
+  context 'creditcard accounts' do
+    before do
+      @ofx = OFX::Parser::Base.new("spec/fixtures/creditcard.ofx")
+      @parser = @ofx.parser
+    end
+
+    it "should set account type" do
+      @parser.account.type.should == :creditcard
+    end
+
```
Comment (iseessel) [inline spec/ofx/ofx102_spec.rb:55]: space

File: spec/ofx/ofx220_spec.rb [hunk 5]
```diff
@@ -64,7 +64,7 @@
         @acct_info.description.should == 'CREDIT CARD'
         @acct_info.bank_id.should == ''
         @acct_info.id.should == '4400720'
-        @acct_info.type.should == nil
+        @acct_info.type.should == :creditcard
```
Comment (iseessel) [inline spec/ofx/ofx220_spec.rb:67]: I see this account type should be credit card: https://github.com/maestrano/ofx/blob/feature/account-discovery/spec/fixtures/v220.ofx#L56.

We may, however, want to add a test case where the account type should be nil, as this `null case` is no longer covered in the specs. e.g. if Account Type is unrecognizable?

**Review by iseessel — APPROVED** [review]
LGTM -- just one suggestion.

**Comment by iseessel** [comment]
Looks like we are having some problems with ruby versions and gems here for the circleCI build. Should we consider bumping ruby version -- not much experience with this. @ouranos

**Comment by ouranos** [comment]
@iseessel we should at some point but we're a bit stuck with the upstream repo. I'll open a PR there and hopefully she can merge that as well as the other ones :)

**Comment by ouranos** [comment]
This has been included in #5
